### PR TITLE
adding haven80 hotswap & haven65 pcbs firmware to qmk

### PR DESCRIPTION
**Pull request by CMMS-Freather**
adds following keyboard firmware to qmk repo
haven80 hotswap
haven65 solder
haven65 hotswap

updated fuji65 via keymap
updated teahouse/ayleen via keymap and config.h to achieve same indicator only control feature in via 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ah/haven80_hotswap/config.h [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// Copyright 2023 Freather
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define EECONFIG_USER_DATA_SIZE 4
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/config.h:6]: Is there a reason for this?

File: keyboards/ah/haven80_hotswap/info.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    "rgblight": {
+        "pin": "E6",
+        "led_count": 21,
+        "hue_steps": 8,
+        "saturation_steps": 8,
+        "brightness_steps": 8,
+        "max_brightness": 255,
+        "sleep": true,
+        "animations": {
+            "all": true
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/info.json:39]:
```suggestion
        "sleep": true,
        "animations": {
            "alternating": true,
            "breathing": true,
            "christmas": true,
            "knight": true,
            "rainbow_mood": true,
            "rainbow_swirl": true,
            "rgb_test": true,
            "snake": true,
            "static_gradient": true,
            "twinkle": true
```

File: keyboards/ah/haven80_hotswap/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    ),
+
+    [1] = LAYOUT(
+        KC_SPC,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    )
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/keymaps/default/keymap.c:24]:
```suggestion
    )
```

File: keyboards/ah/haven80_hotswap/keymaps/vial/rules.mk [hunk 4]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+VIAL_ENABLE = yes
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/keymaps/vial/rules.mk:2]:
```suggestion
```

File: keyboards/ah/haven80_hotswap/readme.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#Haven80 Hotswap
+
+![haven 80 hotswap](https://i.imgur.com/lXKf8Ci.png)
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/readme.md:3]:
```suggestion
![haven 80 hotswap](https://i.imgur.com/lXKf8Cih.png)
```

File: keyboards/ah/haven80_hotswap/readme.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make ah/haven80_hotswap:default
+
+Flashing example for this keyboard:
+
+    make ah/haven80_hotswap:default:flash
+
+
```
Comment (waffle87) [inline keyboards/ah/haven80_hotswap/readme.md:17]:
```suggestion
## Bootloader

```

**Review by waffle87 — CHANGES_REQUESTED** [review]
Not sure why the entire `teahouse/ayleen` directory is being added again, after #19900 was merged?

File: keyboards/ah/haven80_hotswap/config.h (hunk 1 above)
Comment (CMMS-Freather) [reply keyboards/ah/haven80_hotswap/config.h:6]: > Is there a reason for this?

this is for using the eeconfig_read_user_datablock in the via/keymap.c. if this is not included in the config, the corresponding code will not work

File: keyboards/ah/haven80_hotswap/readme.md (hunk 6 above)
Comment (CMMS-Freather) [reply keyboards/ah/haven80_hotswap/readme.md:18]: > 

edited the file to show the corresponding bootloader

**Comment by CMMS-Freather** [comment]
> Not sure why the entire `teahouse/ayleen` directory is being added again, after #19900 was merged?

i am not too sure why either. While making the pr, i only selected files that will be used in the havne80_hotswap.

**Comment by zvecr** [comment]
> i am not too sure why either. While making the pr, i only selected files that will be used in the havne80_hotswap.

Probably because you have workflow issues, most likely due to the reused the branch from a previous PR without fixing the history first.

This will have to be resolved your end before the PR can be accepted.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/ah/haven65_hotswap/config.h [hunk 7]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2023 Freather
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+
+#define EECONFIG_USER_DATA_SIZE 4
```
[the remark this answers is not in the document]
Comment (CMMS-Freather) [reply keyboards/ah/haven65_hotswap/config.h]: > You cant have user config at the keyboard level.

this has been resolved in the newly commited files

**Comment by CMMS-Freather** [comment]
> > i am not too sure why either. While making the pr, i only selected files that will be used in the havne80_hotswap.
> 
> Probably because you have workflow issues, most likely due to the reused the branch from a previous PR without fixing the history first.
> 
> This will have to be resolved your end before the PR can be accepted.

this has been resolved as I had to update some new files for the ayleen to allow the same indicator only control in via. Newly commited files for ayleen will be included

**Review by zvecr — CHANGES_REQUESTED** [review]
* Use of `EECONFIG_KB_DATA_SIZE` at the keymap level is just as bad as `eeconfig_update_kb_datablock` at the keyboard level.
  * My preference would be `EECONFIG_USER_DATA_SIZE` defined at a keymap level `config.h` and `eeconfig_update_user_datablock` calls
* `keyboards/teahouse/ayleen` isnt a set of updates to existing files, as they exist already.
  * The `Files Changed` tab needs to correctly reflect this and trying to re-add them is wrong

image

**Pull request closed without merging**